ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywall.kt

```diff
@@ -175,6 +175,7 @@ internal fun getPaywallViewModel(
 ): PaywallViewModel {
     val applicationContext = LocalContext.current.applicationContext
     val viewModel = viewModel<PaywallViewModelImpl>(
+        key = options.hashCode().toString(),
```

Comment:
This is the main change. With this we can have multiple view models living at the same time.

But more importantly, this is not new but the view model isn't currently cleared when using composables or views and dismissing the paywall, since the view model owner (the activity) isn't cleared. With the increase in the possible number of view models, this means we can potentially increase our memory usage if the paywall is displayed with multiple different options. This should normally not be a problem, but something to consider...

To solve this, it might be tricky to scope the view model to a screen though, since we don't have access to the navigation graph... Any ideas?

Reply:
Yes, if using jetpack navigation. So we should be mostly ok in that case. However, it's not guaranteed for devs to be using that library... But yeah, it further constraints this limitation 👍

Reply:
Yea you're right. It's a good limitation to be aware of!